### PR TITLE
Change grub hdd numbering

### PR DESCRIPTION
**Pull request by mikem-zed**
when grub queries for available disks it doesn't take into account that
the disk can be removable e.g. USB stick. The disk can appear in front of regular
HDDs and the numbering will be different e.g. hd0 become hd1 when the
USB stick is plugged in. It is not a problem for GRUB to find a correct
partition in this case and the system can be booted just fine. However
every command from grub.cfg is measured into PCR8 while being executed
and HDD names appear in those commands e.g. 'set root=(hd2,gpt5)'. If
any key is sealed into TPM using PCR8 then that key cannot be unsealed when a
random USB stick is inserted (or removed if it was inserted when the key
was sealed)

the original issue should not affect PC BIOS case because USB devices
are usually emulated as either CD or floppy drives and have their unique
numbering

**Risk assessment:**

1. 1st and 2nd stages of grub are independent applications. No information is passed between these stages e.g. root drive name. The only purpose of stage 1 is to find and chainload stage 2
2. Stage 2 will find EVE partitions witout knowing how it was loaded by stage 1
3. If 2 stages have different HDD numbering it will not affect eve booting process, however it will affect PCR contents

so there is no ricks of bricking a device with this fix.

### CODE REVIEW

**Comment by mikem-zed** [comment]
I'm currently wating for test results. I already have a version where all formating/printfs are cleaned up just did not wanted to push it in case I still need it. But please look at this now and consider all possible consequences of this fix

**Comment by eriknordmark** [comment]
> **WARNING**: eve cannot be just updated remotely because the numbering of HDDs will be different in 1st and 2nd stages of GRUB

Do they have to be identical? Then I don't understand how we can deploy this fix. We don't have a robust way to update stage1.

File: pkg/grub/patches/0014-Put-removable-hard-drives-detected-by-UEFI-at-the-en.patch [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+diff --git a/grub-core/disk/efi/efidisk.c b/grub-core/disk/efi/efidisk.c
+index 5d2400f66..0260ade55 100644
+--- a/grub-core/disk/efi/efidisk.c
++++ b/grub-core/disk/efi/efidisk.c
+@@ -27,6 +27,8 @@
+ #include <grub/efi/efi.h>
+ #include <grub/efi/disk.h>
+ 
++#define DEBUG_NAMES 1
```
[the remark this answers is not in the document]
Comment (mikem-zed) [reply pkg/grub/patches/0014-Put-removable-hard-drives-detected-by-UEFI-at-the-en.patch:35]: this is why it is marked WIP. I have fixed all these small issues with formating/extra printf but have not pushed. I'm waiting fo test results from CS

**Review by eriknordmark — COMMENTED** [review]
I see a comment from @rene which hasn't been addressed/responded to.

From the description is it clear that we can do this in stage2 independently of what stage1 does? 

With an old stage1 don't we still have an issue that some PCRs change whether or not a USB drive is connected due to measurements from stage1 grub?

**Review by eriknordmark — APPROVED** [review]
Kick off tests

**Pull request merged**